asv_bench/benchmarks/indexing.py

```diff
@@ -83,6 +84,32 @@ def time_loc_slice(self, index, index_structure):
         self.data.loc[:800000]
 
 
+class NumericMaskedIndexing:
+    params = [
+        ("Int64", "UInt64", "Float64"),
+        (True, False),
+    ]
+    param_names = ["dtype", "monotonic"]
+
+    def setup(self, dtype, monotonic):
+        N = 10**6
+        indices = {
+            True: Index(range(N), dtype=dtype),
+            False: Index(
+                list(range(50)) + [54, 53, 52, 51] + list(range(55, N - 1)), dtype=dtype
```

Comment:
NBD but i think if you do this at the class level outside of setup it might only construct once.  might add up

Reply:
Do we run into risks of caching stuff on the Index level that could impact performance?

Reply:
its possible.  could just create the arrays instead of the Index objects

Reply:
Good point, moved the list construction outside. We should probably think about going over the asvs, we are doing this in a lot of places.